### PR TITLE
DistributedApplicationTestingBuilder.CreateAsync - Make an overload public

### LINKED ISSUE #5285: [Aspire.Hosting.Testing] Cannot pass `DistributedApplicationOptions` when using `DistributedApplicationTestingBuilder.CreateAsync`
`DistributedApplicationTestingBuilder.CreateAsync` overload that accepts an Action for the options is
internal.

https://github.com/dotnet/aspire/blob/1f3c78a798a62eac4aca41195f6b3de5d571131a/src/Aspire.Hosting.Testing/DistributedApplicationTestingBuilder.cs#L82-L87

We should consider making this public. One example use of this is to use the `DistributedApplicationOptions` to override the container registry for tests.

cc @mitchdenny @ReubenBond @eerhardt @sebastienros

### PR DESCRIPTION
**Pull request by radical**
.. public. This overload allows configuring
`DistributedApplicationOptions`, and `HostApplicationBuilderSettings`.

```
public static Task<IDistributedApplicationTestingBuilder> CreateAsync<TEntryPoint>(string[] args, Action<DistributedApplicationOptions, HostApplicationBuilderSettings> configureBuilder, Canc>
public static async Task<IDistributedApplicationTestingBuilder> CreateAsync(Type entryPoint, string[] args, Action<DistributedApplicationOptions, HostApplicationBuilderSettings> configureBuil
der, CancellationToken cancellationToken = default)
```

Fixes https://github.com/dotnet/aspire/issues/5285

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/5348)

### CODE REVIEW

File: tests/Aspire.Hosting.Testing.Tests/TestingBuilderTests.cs [hunk 1]
```diff
@@ -13,6 +13,35 @@ namespace Aspire.Hosting.Testing.Tests;
 
 public class TestingBuilderTests
 {
+    [Theory]
+    [RequiresDocker]
+    [InlineData(false)]
+    [InlineData(true)]
+    public async Task CreateAsyncWithOptions(bool genericEntryPoint)
```
Comment (radical) [inline tests/Aspire.Hosting.Testing.Tests/TestingBuilderTests.cs:20]: I'm not sure what a good test might be for this. This is a modification of the existing `HasEndpoints` test.

File: tests/Aspire.Hosting.Testing.Tests/TestingBuilderTests.cs [hunk 2]
```diff
@@ -13,6 +13,34 @@ namespace Aspire.Hosting.Testing.Tests;
...
+    {
+        var nonExistantRegistry = "non-existant-registry-azurecr.io";
+        Action<DistributedApplicationOptions, HostApplicationBuilderSettings> configureBuilder = (options, settings) =>
+        {
+            options.ContainerRegistryOverride = nonExistantRegistry;
```
Comment (radical) [inline tests/Aspire.Hosting.Testing.Tests/TestingBuilderTests.cs:25]: How can I test setting something on `settings`?

File: src/Aspire.Hosting.Testing/DistributedApplicationTestingBuilder.cs [hunk 3]
```diff
@@ -69,6 +69,22 @@ public static Task<IDistributedApplicationTestingBuilder> CreateAsync<TEntryPoin
     public static Task<IDistributedApplicationTestingBuilder> CreateAsync(Type entryPoint, string[] args, CancellationToken cancellationToken = default)
         => CreateAsync(entryPoint, args, (_, __) => { }, cancellationToken);
 
+    /// <summary>
+    /// Creates a new instance of <see cref="DistributedApplicationTestingBuilder"/>.
+    /// </summary>
+    /// <typeparam name="TEntryPoint">
+    /// A type in the entry point assembly of the target Aspire AppHost. Typically, the Program class can be used.
+    /// </typeparam>///
```
Comment (eerhardt) [inline src/Aspire.Hosting.Testing/DistributedApplicationTestingBuilder.cs:77]:
```suggestion
    /// </typeparam>
```

File: tests/Aspire.Hosting.Testing.Tests/TestingBuilderTests.cs (hunk 2 above)
Comment (eerhardt) [reply tests/Aspire.Hosting.Testing.Tests/TestingBuilderTests.cs:25]: Set the `settings.EnvironmentName` to something like `Production` or `Development` or `CustomEnvironment`.

And then verify that `appHost.Environment.EnvironmentName` is set to that name.

**Review by mitchdenny — APPROVED** [review]
LGTM

**Pull request merged**